yarl/__init__.py

```diff
@@ -661,10 +677,7 @@ def _encode_host(cls, host):
             ip, sep, zone = host.partition("%")
             ip = ip_address(ip)
         except ValueError:
-            try:
-                host = idna.encode(host, uts46=True).decode("ascii")
-            except UnicodeError:
-                host = host.encode("idna").decode("ascii")
+            host = _encode_host(host)
```

Comment:
Why not just cache the *whole* `URL._encode_host()` classmethod body here?

I understand that there's a nice symmetry with `_decode_host`, but it is not as if these two new functions are being used in multiple locations. The `URL.host` implementation is pretty thin beyond the `_decode_host()` call and is itself a cached property. In contrast, `URL._encode_host()` adds more string juggling that would benefit from being cached too. The downside would be that the cache would be used for pure ip addresses as well as hostnames.